common/scala/src/main/scala/whisk/core/containerpool/logging/DockerToActivationFileLogStore.scala

```diff
@@ -87,9 +92,11 @@ class DockerToActivationFileLogStore(system: ActorSystem, destinationDirectory:
             val size = element.size
             if (bytesRead + size > maxSize) {
               bytesRead = size
-              val newLogFile = destinationDirectory.resolve(s"userlogs-${Instant.now.toEpochMilli}.log")
-              logging.info(this, s"Rotating log file to '$newLogFile'")
-              Some(newLogFile)
+              val logFilePath = destinationDirectory.resolve(s"userlogs-${Instant.now.toEpochMilli}.log")
+              logging.info(this, s"Rotating log file to '$logFilePath'")
+              Files.createFile(logFilePath, attr)
+              Files.setPosixFilePermissions(logFilePath, perms)
```

Comment:
Both of these can throw an IOException. I'm assuming such an error would cause the wrapping RetrySink to trigger a restart? Should we make sure the message is logged accordingly?

These two lines seem to be redundant, aren't the parameters already passed in the `createFile` call?

Reply:
`createFile` does not properly set the file permissions, relying on `setPosixFilePermissions` instead. 

I threw an error from inside the the `LogRotatorSink` function, it did indeed attempt to recreate the log file.